tests/e2e/pageobjects/cell.page.ts

```diff
@@ -0,0 +1,197 @@
+     * the output has some special formatting. (e.g remove new lines from multine content)
+     * @returns Promise<boolean>
+     */
+    async isSuccessfulExecution(expectedContent?: string): Promise<boolean> {
+        return Boolean(await this.#getCurrentExecutionRow(expectedContent))
+    }
+
+    /**
+     * Checks if the specified output (a string or regular expression) is rendered
+     * @param expectedOutput {string}
+     * @param regex {RegExp}
+     * @returns boolean
+     */
+    async cellOutputExists(expectedOutput: string, expectedTerminal: OutputType, regex?: RegExp): Promise<boolean> {
+        await browser.switchToParentFrame()
+        const notebookIframe = await $('iframe')
+        if (notebookIframe.error) {
+            throw new Error('Could not find notebook iframe')
+        }
+        await browser.switchToFrame(notebookIframe)
+        const executionIFrame = await $('iframe')
+        if (executionIFrame.error) {
+            throw new Error('Could not find execution iframe')
+        }
+        await browser.switchToFrame(executionIFrame)
+        let outputExists = false
+        const rows = await $$(expectedTerminal)
+        for (const row of rows) {
+            if (row.error) {
+                throw row.error
+            }
+            const text = !OutputType.Display ? await row.getText() : await row.getHTML(false)
+            if (regex) {
+                outputExists = regex.test(text)
+            } else if (text.includes(expectedOutput)) {
+                outputExists = true
+            }
+            if (outputExists) {
+                break
+            }
+        }
+        await browser.switchToParentFrame()
+        await browser.switchToParentFrame()
+        return outputExists
+    }
+
+    /**
+     * Access the terminal associated and retrieve the complete text that is 
+     * displayed in the terminal window.
+     * @param openTerminal {boolean} Indicate if the terminal should be opened (default: true)
+     * @returns 
+     */
+    async getTerminalText(openTerminal: boolean = true) {
+        const workbench = await browser.getWorkbench()
+        if (openTerminal) {
+            await this.openTerminal()
+        }
+        await workbench.executeCommand('Terminal select all')
+        await workbench.executeCommand('Copy')
+        const text = await clipboard.read()
+        await clipboard.write('')
+        await workbench.executeCommand('kill all terminals')
+        return text
+    }
+
+    /**
+     * Stalls until the success status check is detected
+     * @param timeout {number} The maximum duration for waiting until it times out
+     * @param expectedContent {string} Override the cell text value in case
+     * the output has some special formatting. (e.g remove new lines from multine content)
+     */
+    async waitForSuccess(expectedContent?: string | undefined, timeout = DEFAULT_SEARCH_FOR_CELL_TIMEOUT) {
```

Comment:
maybe have a more specific name like
```suggestion
    async waitForCellOutput(expectedContent?: string | undefined, timeout = DEFAULT_SEARCH_FOR_CELL_TIMEOUT) {
```